trillian/ctfe/instance.go

```diff
@@ -63,15 +63,27 @@ type InstanceOptions struct {
 }
 
 // SetUpInstance sets up a log (or log mirror) instance using the provided
-// configuration, and returns a set of handlers for this log.
+// configuration, and returns a set of handlers for this log. This is the
+// standard way of setting up a log instance.
 func SetUpInstance(ctx context.Context, opts InstanceOptions) (*PathHandlers, error) {
+	handlers, _, err := SetUpInstance2(ctx, opts)
+	return handlers, err
+}
+
+// SetUpInstance2 sets up a log (or log mirror) instance using the provided
+// configuration, and returns a set of handlers for this log and the
+// STHGetter being used for it.
+//
+// Note: This is for experiments with DNS serving and is not currently used
+// by CTFE. This may either go away or become an official thing in future.
+func SetUpInstance2(ctx context.Context, opts InstanceOptions) (*PathHandlers, STHGetter, error) {
```

Comment:
Could maybe make this the one and only `SetUpInstance`, and then:
 - merge `ctfe.PingTreeHead()` & `LogSTHGetter.GetSTH()` into a single method
 - modify `ct_server/main.go` to use the returned `STHGetter` rather than `ctfe.PingTreeHead`

Or not, if this is just for experimentation.  But a better name (e.g. `SetUpInstanceWithSTHGetter`) would be nice even for an experiment -- names like `Thing2` and `WhatsitNew` have a habit of living on longer than anticipated...

Reply:
I've gone with SetUpInstanceAndGetter. I don't want to modify the main code yet until it's clear where this is going.